AEPLifecycle/Sources/LifecycleV2/LifecycleV2.swift

```diff
@@ -53,7 +57,20 @@ class LifecycleV2 {
         stateManager.update(state: .START) { [weak self] (updated: Bool) in
             guard let self = self else { return }
             guard updated else { return }
-            // todo: MOB-14878 create launch application event and if needed the close event
+
+            // detect a possible crash/incorrect start/pause implementation
+            if !isInstall && self.isCloseUnknown(prevAppStart: self.dataStoreCache.getAppStartDate(), prevAppPause: self.dataStoreCache.getAppPauseDate()) {
+                // in case of an unknown close situation, use the last known app close event timestamp
+                guard let xdm = self.xdmMetricsBuilder.buildAppCloseXDMData(launchDate: self.dataStoreCache.getAppStartDate() ?? Date(), closeDate: self.dataStoreCache.getCloseDate() ?? Date(), isCloseUnknown: false) else { return }
```

Comment:
isCloseUnknown: true

Reply:
I would prefer changing the logic to 
if let xdm = self.xdmMetricsBuilder.buildAppCloseXDMData(...) { 
 self.dispatchApplicationClose(xdm: xdm)
}
instead of returning early.

Reply:
+1 and can we use the same for buildAppLaunchXDMData and inside pause

Reply:
Few other updates required here:
- buildAppCloseXDMData should be updated and include the fallbackCloseEventDate (see Android implementation)
- we should not pass current Date() when app start/app pause are nil, we should handle the case accordingly (see Android computeSessionLength where it checks for both start/pause to be valid values).